### ISSUE TITLE
Feature Request: filament change at layer height for non-ams

### ISSUE THREAD

**Issue by CruxMason** [bambulab/BambuStudio#320]
In addition to the pause at layer add a color change where the machine will complete an unload and then wait for filament again and do the purge and poop before starting.  Prusa slicer and the mk3s does this well.

**Comment by YifanWuBambu** [comment]
Hi @CruxMason ,
Do you mean this feature?
![image](https://user-images.githubusercontent.com/110661856/192091116-e36361d6-b6a9-42eb-80af-e19884b02b33.png)

**Comment by CruxMason** [comment]
Yes

**Comment by whiiiskyy** [comment]
Yes. This will also require firmware change.

IIRC, UncleJessy on YouTube showed the unload filament is gray out.
Apparently at this stage, only with AMS you can change colors.

This feature should be included not just because of color change but many of us pause the printer to insert nuts/magnets before resuming it so "Pause at layer X" is a must have feature too.

**Comment by kiliankoe** [comment]
This specific feature also does not work *with* the AMS. It's completely ignored by the printer. The only way to currently get a filament change is by using the painting features. Support for this layer-based approach would be much appreciated.

<img width="289" alt="image" src="https://user-images.githubusercontent.com/2625584/192161952-4e250f85-5030-4167-ab17-8c5113303c89.png">

**Comment by YifanWuBambu** [comment]
> Yes. This will also require firmware change.
> 
> IIRC, UncleJessy on YouTube showed the unload filament is gray out. Apparently at this stage, only with AMS you can change colors.
> 
> This feature should be included not just because of color change but many of us pause the printer to insert nuts/magnets before resuming it so "Pause at layer X" is a must have feature too.

![image](https://user-images.githubusercontent.com/110661856/192175625-8629d9d6-0152-4694-bc77-5f534f8eee15.png)

**Comment by YifanWuBambu** [comment]
> This specific feature also does not work _with_ the AMS. It's completely ignored by the printer. The only way to currently get a filament change is by using the painting features. Support for this layer-based approach would be much appreciated.
> 
> <img alt="image" width="289" src="https://user-images.githubusercontent.com/2625584/192161952-4e250f85-5030-4167-ab17-8c5113303c89.png">

We will check this issue. Thanks for reporting that.

**Comment by VictorGengGC** [comment]
This issue has been inactive for a long time, so we will close it for now. If you are still able to reproduce the problem, please feel free to open a new issue or submit a support ticket so we can continue investigating. Sorry for the inconvenience.
